### DISCUSSION TITLE
Unable to connect to existing server address

### DISCUSSION THREAD

**Discussion by firmansyahrosid** [webdriverio/webdriverio#12453] [Q&A] [answered]
I run appium server at address 172.10.10.1 but when I check webdriverio logs, it always tries to connect the driver at address 127.0.0.1.

this is my wdio configuration.
```
baseConfig.services = [
  ...baseConfig.services,
  [
    "appium",
    {
      command: "appium",
      args: {
        address: "172.10.10.1",
        callbackAddress: "172.10.10.1",
      } as AppiumServerArguments,
    },
  ],
];
```

and here is the wdio log.

```
[0-25] 2024-03-08T01:33:23.651Z DEBUG @wdio/runner: init remote session
[0-25] 2024-03-08T01:33:23.651Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-25] 2024-03-08T01:33:23.651Z INFO @wdio/utils: Connecting to existing driver at http://127.0.0.1:4723/
[0-25] 2024-03-08T01:33:23.655Z INFO webdriver: [POST] http://127.0.0.1:4723/session
```

**Comment by christian-bromann** [comment]
Can you try to set `172.10.10.1` to the [`hostname`](https://webdriver.io/docs/configuration#hostname) option?

**Reply by firmansyahrosid** [reply]
still having the same problem. Am I doing it wrong?

<img width="848" alt="image" src="https://github.com/webdriverio/webdriverio/assets/102026790/441fe482-4a59-48d3-984a-5de62405e617">

**Reply by christian-bromann** [reply]
Can you please provide the full logs?

**Reply by firmansyahrosid** [reply]
Hi @christian-bromann Appium is running globally on IP 192.168.0.5 and I set the address/hostname in wdio to be the same as Appium IP and here are the logs.
https://gist.github.com/firmanrosid/23abcc906f7c1d521cfbaac69beed692

**Reply by christian-bromann** [reply]
It seems like Appium fails to start due to:
> The requested port may already be in use. Please make sure there is no other instance of this server running already.

**Reply by firmansyahrosid** [reply]
So what if I want to use an existing server?

**Reply by christian-bromann** [reply, marked as answer]
Then you don't need a service for starting Appium 😉 Just connect to the server using the `hostname` and `port` options.

**Reply by firmansyahrosid** [reply]
thankyou @christian-bromann
